src/verifier.rs

```diff
@@ -11,14 +12,16 @@ use crate::{blake_hash_usize_to_curve, fft_precompute, msm_execute_parallel, msm
 
 pub const SECURITY_BITS: usize = 128;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct VerificationKey<C: HaloCurve> {
     pub c_constants: Vec<AffinePoint<C>>,
     pub c_s_sigmas: Vec<AffinePoint<C>>,
     pub degree: usize,
     pub num_public_inputs: usize,
     pub security_bits: usize,
+    #[serde(skip)]
```

Comment:
Oh yes you're right, I was a bit lazy with those. I can replace `#[serde(skip)]` with `#[serde(skip_serializing_if = "Option::is_none")]` and then add a method to change these attributes to `None` if someone wants to save space when serializing a vk?
After a quick test, it seems like I would get into the same trouble as in https://github.com/mir-protocol/plonky/pull/82#discussion_r494839509 with `OpeningSet<F: Field>` for `FftPrecomputation<F: Field>`. But this can be fixed by defining a new type `FftPrecomputationCurve<C: Curve> = FftPrecomputation<C::ScalarField>`. Would that work for you?

Reply:
`#[serde(skip_serializing_if = "Option::is_none")]` sounds like a neat solution!

Hm I'd prefer to keep `FftPrecomputation<F: Field>`, but I'll take a look at that error now and see if we can work around it.

Reply:
Oh yeah that error seems tricky. Feel free to merge like this and I'll look into it more later.

Reply:
Yeah I spent quite some time trying to solve it for `OpeningSet` but without success. Maybe I'll try asking on Stackoverflow with a minimal failing example. 
I'll merge now and try to come back later with a fix.